### PR TITLE
⬆️ #2299 - Upgrade dependencies

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

Update dependencies to prepare for updating to React 17.

- Significant updates
  - Next 10 
  - Typescript 4
  - Husky 4
- Remove
  - recompose - barely used
  - types/graphql - stub library, types are in graphql library
  - immer - only in "resolutions", might be a vestigial dependency
  - jwt-decode - not used, we use Ego utils
  - react-docgen-typescript - old storybook dependency, not used
  - gitmoji-cli - had a vulnerable dependency, not needed
- Other package updates are in the commit messages below
- Some changes were needed due to Typescript update
- Moved some dependencies to dev dependencies
- Dockerfile update: add Uikit build step because, after updates, Uikit components weren't being found during build.

## Type of Change

- [ ] Bug
- [ ] Styling
- [x] New Feature

## Checklist before requesting review

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- UI Kit (select one):
  - [ ] Modified Existing components
  - [x] No new UIKit components or changes
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2299
https://github.com/icgc-argo/platform-ui/issues/2299

### CODE REVIEW

**Comment by ciaranschutte** [comment]
@samrichca can we leave out the uikit changes please? I'm trying to get that one wrapped today. It's taking a long time because there's been additions here that were not synced. So this will just add more to it.

**Comment by samrichca** [comment]
Looked over `resolutions` in package.json and I'll have to revisit them when I get to the Babel ticket at the end. Made a note of that in my spreadsheet.

**Comment by ciaranschutte** [comment]
noticed `husky 4` is a significant update.
worth upgrading to the latest 8 if we're upgrading at all? https://github.com/typicode/husky @samrichca

**Comment by samrichca** [comment]
@ciaranschutte I actually did the Husky 4 upgrade because I couldn't commit to this branch without it. I'm planning on doing a Husky 8 and Prettier upgrade and running Prettier on the codebase on that PR. Does that work?

**Comment by ciaranschutte** [comment]
> @ciaranschutte I actually did the Husky 4 upgrade because I couldn't commit to this branch without it. I'm planning on doing a Husky 8 and Prettier upgrade and running Prettier on the codebase on that PR. Does that work?

sounds good

File: package.json [hunk 1]
```diff
@@ -3,15 +3,12 @@
   "version": "1.110.1",
   "scripts": {
     "dev": "next -p 8080",
+    "local-build": "rm -rf .next && next build",
```
Comment (ciaranschutte) [inline package.json:6]: same as `build` command? @samrichca

**Review by ciaranschutte — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -3,11 +3,10 @@
   "version": "1.108.0",
   "scripts": {
     "dev": "next -p 8080",
+    "local-build": "npm run build-uikit && rm -rf .next && next build",
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply package.json]: ^outdated

File: package.json (hunk 1 above)
Comment (samrichca) [reply package.json:6]: Doh, good catch. Added this when uikit was giving me trouble.

Comment (samrichca) [reply package.json]: Removed

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**